programs/survival/lava_sponge.sc

```diff
@@ -0,0 +1,43 @@
+//just a sponge work on lava like water
+//by _GieR
+
+
+__config() -> (
+   m(
+      l('stay_loaded','true')
+   )
+);
+
+
+__on_player_places_block(player, item_tuple, hand, block)->(
+    if(hand != 'mainhand' || item_tuple:0 != 'sponge' || all(neighbours(pos(block)),_!='lava'), return());
+    set(pos(block),'wet_sponge');
+    count = 0;
+    recursion = 0;
+    sponge(l(pos(block)), recursion, count);
+);
+
+sponge(coord_list, recursion, count)->(
+    lava = {};
+    for(coord_list,
+        coord = _;
+        for(neighbours(coord), 
+            if(!has(lava, pos(_)) && recursion<=7 && count<=64,

```

Comment:
could separate the check for recursion to be at the top of the function to avoid repetive checks, and the check for count to be in an if statement outside this one, so that if it fails, you can break out of the loop, and save some compute time in looping needlessly and checking the same condition over and over.

Reply:
Do you mean a structure like `if recursion( if count ( if has(lava) ( if block == 'lava')))) ` ?

Reply:
no. I mean `if(count etc, if(lava), break())`, and put the recursion check outside the loop cos the recursion variable is updated outside the loop, so checking it over and over again is effectively a waste of time. Generally with recursion you check the recursive depth at the top of the function.

Reply:
My idea was recursion and count are just simple int check, so i put them together to get prettier code look

Reply:
plz check the latest commit